skimage/morphology/max_tree.py

```diff
@@ -0,0 +1,669 @@
+    ----------
+    image: ndarray
+        The input image for which the max-tree is to be calculated.
+        This image can be of any type.
+    connectivity: unsigned int, optional
+        The neighborhood connectivity. The integer represents the maximum
+        number of orthogonal steps to reach a neighbor. In 2D, it is 1 for
+        4-connectivity and 2 for 8-connectivity. Default value is 2.
+
+    Returns
+    -------
+    parent: ndarray, int64
+        The value of each pixel is the index of its parent in the ravelled
+        array
+
+    tree_traverser: 1D array, int64
+        The ordered pixel indices (referring to the ravelled array). The pixels
+        are ordered such that every pixel is preceded by its parent (except for
+        the root which has no parent).
+
+    References
+    ----------
+    .. [1] Salembier, P., Oliveras, A., & Garrido, L. (1998). Antiextensive
+           Connected Operators for Image and Sequence Processing.
+           IEEE Transactions on Image Processing, 7(4), 555-570.
+           DOI:10.1109/83.663500
+    .. [2] Berger, C., Geraud, T., Levillain, R., Widynski, N., Baillard, A.,
+           Bertin, E. (2007). Effective Component Tree Computation with
+           Application to Pattern Recognition in Astronomical Imaging.
+           In International Conference on Image Processing (ICIP) (pp. 41-44).
+           DOI:10.1109/ICIP.2007.4379949
+    .. [3] Najman, L., & Couprie, M. (2006). Building the component tree in
+           quasi-linear time. IEEE Transactions on Image Processing, 15(11),
+           3531-3539.
+           DOI:10.1109/TIP.2006.877518
+    .. [4] Carlinet, E., & Geraud, T. (2014). A Comparative Review of
+           Component Tree Computation Algorithms. IEEE Transactions on Image
+           Processing, 23(9), 3885-3895.
+           DOI:10.1109/TIP.2014.2336551
+
+    Examples
+    --------
+    We create a small sample image (Figure 1 from [4]) and build the max-tree.
+
+    >>> image = np.array([[15, 13, 16], [12, 12, 10], [16, 12, 14]])
+    >>> P, S = max_tree(image, connectivity=2)
+    """
+    # User defined masks are not allowed, as there might be more than one
+    # connected component in the mask (and therefore not a single tree that
+    # represents the image). Mask here is an image that is 0 on the border
+    # and 1 everywhere else.
+    mask = np.ones(image.shape)
+    for k in range(len(shape)):
+        np.moveaxis(mask, k, 0)[0] = 0
+        np.moveaxis(mask, k, 0)[-1] = 0
+
+    neighbors, offset = _validate_connectivity(image.ndim, connectivity,
+                                               offset=None)
+
+    # initialization of the parent image
+    parent = np.zeros(image.shape, dtype=np.int64)
+
+    # flat_neighborhood contains a list of offsets allowing one to find the
+    # neighbors in the ravelled image.
+    flat_neighborhood = _compute_neighbors(image, neighbors,
+                                           offset).astype(np.int32)
```

Comment:
This was renamed to `_offsets_to_raveled_neighbors` in #3022. You'll need to change `image` to `image.shape` as well.

Reply:
OK, this I did now, thanks a lot.